Src/Couchbase.Tests/Core/ClusterControllerTests.cs

```diff
@@ -178,6 +178,35 @@ public void When_Same_Bucket_Requested_Twice_In_Parallel_Only_One_Bootstrap_Is_D
             Assert.IsNotNull(bucket1);
             Assert.AreSame(bucket1, bucket2);
         }
+
+        [Test]
+        public async Task When_Disposing_Bucket_In_Parallel_Does_Not_Dispose_Referenced_Bucket()
```

Comment:
In the previous version of the code, this unit test will throw an ObjectDisposedException when one thread disposes the bucket another thread has been given.